File: front/src/app/_services/sujet.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from 'src/environments/environment';
import { Message } from '../models/message.entity';
import { Sujet } from '../models/sujet.entity';

@Injectable({
  providedIn: 'root',
})
export class SujetService {
  constructor(private http: HttpClient) {}

  findAll() {
    return this.http.get<Sujet[]>(`${environment.apiBaseUrl}/sujets`);
  }

  findById(id: string) {
    return this.http.get<Sujet>(`${environment.apiBaseUrl}/sujets/${id}`);
  }

  create(sujet: Sujet) {
    return this.http.post<Sujet>(`${environment.apiBaseUrl}/sujets`, sujet);
  }

  delete(id: string) {
    return this.http.delete(`${environment.apiBaseUrl}/sujets/${id}`);
  }

  update(sujet: Sujet) {
    return this.http.put<Sujet>(`${environment.apiBaseUrl}/sujets`, sujet);
  }
  like(id: string) {
    return this.http.get(`${environment.apiBaseUrl}/sujets/${id}/like`);
  }
  unlike(id: string) {
    return this.http.get(`${environment.apiBaseUrl}/sujets/${id}/unlike`);
  }

  postMessage(id: string, message: Message) {
    return this.http.post(
      `${environment.apiBaseUrl}/sujets/${id}/messages`,
      message
    );
  }
}

File: back1/src/main/java/com/nico/back/services/SujetService.java
package com.nico.back.services;

import com.nico.back.models.Message;
import com.nico.back.models.Sujet;
import com.nico.back.repositories.SujetRepository;

import java.util.List;

public class SujetService {

    private final SujetRepository sujetRepository;


    public SujetService(SujetRepository sujetRepository) {
        this.sujetRepository = sujetRepository;
    }


    public List<Sujet> findAll() {

        return this.sujetRepository.findAll();
    }

    public Sujet findById(String id) {
        return this.sujetRepository.findById(id).get();
    }


    public Sujet create(Sujet entity) {
        return this.sujetRepository.save(entity);
    }


    public Sujet update(Sujet entity) {
        return this.sujetRepository.save(entity);
    }


    public void deleteById(String id) {
        this.sujetRepository.deleteById(id);
    }

    public void like(String id){
        Sujet sujet = this.sujetRepository.findById(id).get();
        sujet.setLikes(sujet.getLikes()+1);
        this.sujetRepository.save(sujet);
    }
}
